inc/azure_c_shared_utility/umock_c_prod.h

```diff
@@ -10,6 +10,12 @@
     by adjusting the include paths).
 */
 
+#if defined(_DLL) && (defined(_WIN32) || defined(_WIN64))
+#define DLLEXPORT __declspec(dllexport)
+#else
+#define DLLEXPORT
+#endif
```

Comment:
This is the wrong place to do this. umock should not care about DLLEXPORT.

Reply:
As a matter of fact any changes to this file should be reverted. Normally the C shared and IoT SDK should use the umock_c_prod header from umock. But because MBED this shortcut exists and it never got cleaned up. The point is you dont want to have this file diverging from the one in umock_c.

---
In reply to: [184153082](https://github.com/Azure/azure-c-shared-utility/pull/152#discussion_r184153082) [](ancestors = 184153082)

Reply:
I could suggest having a common header called "dll_export.h" :-) which should contain the define et all

---
In reply to: [184153426](https://github.com/Azure/azure-c-shared-utility/pull/152#discussion_r184153426) [](ancestors = 184153426,184153082)

Reply:
Agreed. This was for the purposes of testing. I had considered if this functionality could be incorporated into the MOCKABLE_FUNCTION macro but decided better of it. I have now defined it in macro_utils.h which seemed an appropriate place. Happy to put it in its own file though if you so desire. I'll get rid of the commits in this file before I send it to the build gate.

---
In reply to: [184154823](https://github.com/Azure/azure-c-shared-utility/pull/152#discussion_r184154823) [](ancestors = 184154823,184153426,184153082)